tests/utils/stl/test/Test.py

```diff
@@ -0,0 +1,203 @@
+        return self.path_in_suite[-2]
+
+    def getExecDir(self):
+        return self.getOutputDir()
+
+    def getExecPath(self):
+        return self.getExecDir() / (self.getOutputBaseName() + '.exe')
+
+    def getFullName(self):
+        return '/'.join((Test.getFullName(self), str(self.env_num)))
+
+    def getPassFailResultCodes(self):
+        should_fail = self.isExpectedToFail()
+        pass_var = XPASS if should_fail else PASS
+        fail_var = XFAIL if should_fail else FAIL
+
+        return pass_var, fail_var
+
+    def setup(self):
+        exec_dir = self.getExecDir()
+        source_dir = Path(self.getSourcePath()).parent
+
+        shutil.rmtree(exec_dir, ignore_errors=True)
+        exec_dir.mkdir(parents=True, exist_ok=True)
+
+        for path in source_dir.iterdir():
+            if path.is_file() and path.name.endswith('.dat'):
+                os.link(path, exec_dir / path.name)
+
+    def cleanup(self):
+        shutil.rmtree(self.getExecDir(), ignore_errors=True)
+
+    def getXMLOutputTestName(self):
+        return '/'.join((self.path_in_suite[-2], str(self.env_num)))
+
+    def getXMLOutputClassName(self):
+        safe_test_path = [x.replace(".", "_") for x in self.path_in_suite[:-1]]
+        safe_suite_name = self.suite.name.replace(".", "-")
+
+        if safe_test_path:
+            return safe_suite_name + "." + "/".join(safe_test_path)
+        else:
+            return safe_suite_name + "." + safe_suite_name
+
+    def _configure_cxx(self, lit_config, envlst_entry, default_cxx):
+        env_compiler = envlst_entry.getEnvVal('PM_COMPILER', 'cl')
+
+        if not os.path.isfile(env_compiler):
+            cxx = _compiler_path_cache.get(env_compiler, None)
+
+            if cxx is None:
+                search_paths = self.config.environment['PATH']
+                cxx = shutil.which(env_compiler, path=search_paths)
+                _compiler_path_cache[env_compiler] = cxx
+
+        if not cxx:
+            lit_config.fatal('Could not find: %r' % env_compiler)
+
+        flags = list()
+        compile_flags = list()
+        link_flags = list()
+
+        flags.extend(default_cxx.flags or [])
+        compile_flags.extend(default_cxx.compile_flags or [])
+        link_flags.extend(default_cxx.link_flags or [])
+
+        flags.extend(envlst_entry.getEnvVal('PM_CL', '').split())
+        link_flags.extend(envlst_entry.getEnvVal('PM_LINK', '').split())
+
+        if ('clang'.casefold() in os.path.basename(cxx).casefold()):
+            target_arch = self.config.target_arch.casefold()
+            if (target_arch == 'x64'.casefold()):
+                compile_flags.append('-m64')
+            elif (target_arch == 'x86'.casefold()):
+                compile_flags.append('-m32')
+
+        self.cxx = CXXCompiler(cxx, flags, compile_flags, link_flags,
+                               default_cxx.compile_env)
+
+    # This is mostly lifted from lit's test class. The changes here are to

```

Comment:
It sounds like this file should have LLVM's banner then.

Reply:
Yeah, wasn't sure how to handle that. Everything else in the file is mine but that one function is pretty much from lit.

Reply:
Maybe I should isolate that code somewhere else?